### DISCUSSION TITLE
Adding more variables to e3sm_diags mvm

### DISCUSSION THREAD

**Discussion by mahf708** [E3SM-Project/e3sm_diags#674] [Q&A] [answered]
In the case of model-vs-obs, it makes sense that we are somewhat rigid about what we can do, e.g. there are no experimental/observational data to compare to.

However, in the case of model-vs-model, is there a way to add a parameter of interest? Obviously said parameter will likely have no observational counterpart, but will definitely have a model counterpart. For now, I am actually simply interested in adding the `_d1` suffixed versions of these two parameters below (FSNTOA and FSNTOAC). Can I simply specify `variables = [... , 'FSNTOA_d1', 'FSNTOAC_d1']`? If so, I actually did that in the upper block (default.vars) but that didn't seem to work, so I may need to add it under `e3sm_diags.atm_monthly_180x360_aave_mvm.vars` or `e3sm_diags.atm_monthly_180x360_aave_mvm.variables`?

Thank you!

```
[#]
sets = ["lat_lon"]
case_id = "model_vs_model"
variables = ["FSNTOA"]
seasons = ["ANN", "DJF", "MAM", "JJA", "SON"]
contour_levels = [50, 75, 100, 125, 150, 175, 200, 225, 250, 275, 300, 325, 350]
diff_levels = [-30, -25, -20, -15, -10, -5, -2, 2, 5, 10, 15, 20, 25, 30]


[#]
sets = ["lat_lon"]
case_id = "model_vs_model"
variables = ["FSNTOAC"]
seasons = ["ANN", "DJF", "MAM", "JJA", "SON"]
contour_levels = [50, 75, 100, 125, 150, 175, 200, 225, 250, 275, 300, 325, 350, 375]
diff_levels = [-30, -25, -20, -15, -10, -5, -2, 2, 5, 10, 15, 20, 25, 30]
```

https://github.com/E3SM-Project/e3sm_diags/blob/8ea5bad5f553a09b099df9464471b424eaa56b0c/e3sm_diags/driver/default_diags/lat_lon_model_vs_model.cfg#L78-L93

**Comment by mahf708** [comment]
Also, why are the contour levels slightly different above? Or am I not reading this correctly?

**Comment by mahf708** [comment]
I also looked here, but I couldn't find anything about variables: https://github.com/E3SM-Project/zppy/blob/cdd11e983a52ca54e81168de199be611e42d49f8/zppy/templates/default.ini#L100-L236. I only found this https://e3sm-project.github.io/e3sm_diags/_build/html/main/available-parameters.html but I don't know how that translates to a variable inside a .cfg file...

**Comment by mahf708** [comment]
Update: trying adding the variables didn't work :(

**Comment by chengzhuzhang** [comment, marked as answer]
@mahf708 firstly, the variables need to be added to zppy's climo task configuration, by extending the variable list: https://github.com/E3SM-Project/zppy/blob/cdd11e983a52ca54e81168de199be611e42d49f8/zppy/templates/default.ini#L48. 

Re: adding variables in e3sm_diags runs, here is the instruction on how to add new variables on-the-fly in standalone e3sm diags runs:
https://e3sm-project.github.io/e3sm_diags/_build/html/main/add-new-diagnostics.html

However, I think you will need to include code change in zppy to do model vs model in zppy with additional variables.

**Reply by mahf708** [reply]
Thanks, @chengzhuzhang, yes, the climo additions work for me :)

I was hoping I would add them on the fly to e3sm_diags, but I don't think that's possible yet. Thanks a lot for the quick response!
